src/book/eventsFetcher.service.js

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable class-methods-use-this */
+
+const moment = require('moment');
+const { reduce } = require('asyncro');
+
+const { errors } = require('./book.errors');
+const { redisEventsKey } = require('../../config/book.config.json');
+const { assertOrThrow } = require('../../utils');
+
+class EventsFetcherService {
+  constructor(clientExtractor, roomResources, redisClient, usersMapper) {
+    this.clientExtractor = clientExtractor;
+    this.roomResources = roomResources;
+    this.redis = redisClient;
+    this.usersMapper = usersMapper;
+  }
+
+  getRoomEmails() {
+    const roomEmails = [];
+    this.roomResources.forEach((room) => {
+      roomEmails.push(room.email);
+    });
+
+    return roomEmails;
+  }
+
+  minutesToStart(date) {
+    const now = moment();
+    const end = moment(date);
+    const duration = moment.duration(end.diff(now));
+    const diffMins = Math.round(duration.asMinutes());
+
+    return diffMins;
+  }
+
+  async getAllEvents() {
+    const client = await this.getClient();
+    const res = await client.events.list({
+      calendarId: process.env.BOT_EMAIL,
+      timeMin: moment(),
```

Comment:
Shouldn't it be `moemnt().toDate()`?
```suggestion
      timeMin: moment().toDate(),
```

Reply:
Good catch